### PR TITLE
feat(missing_path_parameter): replace rule with spectral:oas rules

### PR DESCRIPTION
**Pull request by padamstx**
## PR summary
This commit removes the legacy 'missing_path_parameter" rule
and instead enables the spectral:oas 'path-params' and
'path-declarations-must-exist' rules, which provide equivalent
functionality.

### General checklist
Please make sure that your PR fulfills the following requirements:  
- [x] The commit message follows the [Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] Dependencies have been updated as needed
- [ ] .secrets.baseline updated as needed?

#### Checklist for adding a new validation rule:
- [x] Added new validation rule definition (packages/ruleset/src/rules/*.js, index.js)
- [ ] If necessary, added new validation rule implementation (packages/ruleset/src/functions/*.js, updated index.js)
- [x] Added new rule to default configuration (packages/ruleset/src/ibm-oas.js)
- [ ] Added tests for new rule (packages/ruleset/test/*.test.js)
- [x] Added docs for new rule (docs/ibm-cloud-rules.md)

#### Checklist for removing an old validation rule:
- [x] Removed old rule implementation (packages/validator/src/plugins/validation/*.js)
- [x] Removed or adjusted testcases (packages/validator/test)
- [x] Updated default configuration to deprecate old rule (packages/validator/src/.defaultsForValidator.js)
- [ ] Removed docs of old rule (docs/ibm-legacy-rules.md)

### CODE REVIEW

File: packages/validator/src/.defaultsForValidator.js [hunk 1]
```diff
@@ -102,6 +101,7 @@ const deprecated = {
   'no_request_body_name': 'request-body-name (spectral rule)',
   'operation_id_case_convention': 'operation-id-case-convention (spectral rule)',
   'operation_id_naming_convention': 'operation-id-naming-convention (spectral rule)',
+  'missing_path_parameter': 'path-params and path-declarations-must-exist (spectra rule)',
```
Comment (pyrooka) [inline packages/validator/src/.defaultsForValidator.js:104]:
```suggestion
  'missing_path_parameter': 'path-params and path-declarations-must-exist (spectral rule)',
```

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js [hunk 2]
```diff
@@ -89,17 +89,17 @@ module.exports.validate = function({ resolvedSpec }, config) {
...
+          // if (checkStatus != 'off') {
+          //   missingParameters.forEach(name => {
+          //     messages.addMessage(
+          //       ['paths', pathName, opName, 'parameters'],
+          //       `Operation must include a path parameter with name: ${name}.`,
+          //       checkStatus,
+          //       'missing_path_parameter'
+          //     );
+          //   });
+          // }
```
Comment (pyrooka) [inline packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js:102]: I guess this should be removed, right?

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js [hunk 3]
```diff
@@ -113,17 +113,17 @@ module.exports.validate = function({ resolvedSpec }, config) {
...
+          // if (checkStatus != 'off') {
+          //   missingParameters.forEach(name => {
+          //     messages.addMessage(
+          //       ['paths', pathName],
+          //       `Path parameter must be defined at the path or the operation level: ${name}.`,
+          //       checkStatus,
+          //       'missing_path_parameter'
+          //     );
+          //   });
+          // }
```
Comment (pyrooka) [inline packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js:126]: Same question as above.

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js [hunk 4]
```diff
@@ -104,19 +104,19 @@ module.exports.validate = function({ resolvedSpec }) {
...
+    //     parameterDefinition.in === 'path' &&
+    //     pathTemplates.indexOf(parameterDefinition.name) === -1
+    //   ) {
+    //     messages.addMessage(
+    //       parameterDefinition.$$path || `paths.${pathName}.parameters[${i}]`,
+    //       `Path parameter was defined but never used: ${parameterDefinition.name}`,
+    //       'error'
+    //     );
+    //   }
+    // });
```
Comment (pyrooka) [inline packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js:119]: Same as above-above. :)

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js [hunk 5]
```diff
@@ -131,17 +131,17 @@ module.exports.validate = function({ resolvedSpec }) {
...
+      //   each(availableParameters, (parameterDefinition, i) => {
+      //     // Assertation 1, for cases when no templating is present on the path
+      //     if (parameterDefinition.in === 'path') {
+      //       messages.addMessage(
+      //         `paths.${pathName}.parameters[${i}]`,
+      //         `Path parameter was defined but never used: ${parameterDefinition.name}`,
+      //         'error'
+      //       );
+      //     }
+      //   });
```
Comment (pyrooka) [inline packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js:144]: Same as above-above-above. :)

**Review by pyrooka — APPROVED** [review]
Just a few comments, but it looks good overall!

File: packages/ruleset/src/ibm-oas.js [hunk 6]
```diff
@@ -3,6 +3,8 @@ const { oas } = require('@stoplight/spectral-rulesets');
 const ibmRules = require('./rules');
 module.exports = {
   extends: oas,
+  documentationUrl:
+    'https://github.com/IBM/openapi-validator/blob/main/docs/ibm-cloud-rules.md',
```
Comment (dpopp07) [inline packages/ruleset/src/ibm-oas.js:7]: 👍

**Review by dpopp07 — COMMENTED** [review]
+1 to all of Norbert's comments but the change looks good otherwise! 👍

File: packages/validator/src/.defaultsForValidator.js (hunk 1 above)
Comment (padamstx) [reply packages/validator/src/.defaultsForValidator.js:104]: fixed

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js (hunk 2 above)
Comment (padamstx) [reply packages/validator/src/plugins/validation/2and3/semantic-validators/paths-ibm.js:102]: yep, I intended to take another pass through the changes in the legacy rule and remove these commented-out sections.  will fix.

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js (hunk 5 above)
Comment (padamstx) [reply packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js:144]: fixed

File: packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js (hunk 4 above)
Comment (padamstx) [reply packages/validator/src/plugins/validation/2and3/semantic-validators/paths.js]: fixed

**Pull request merged**